js/src/admin/components/AuthPage.js

```diff
@@ -0,0 +1,215 @@
+import Page from './Page';
+import Checkbox from '../../common/components/Checkbox';
+import Button from '../../common/components/Button';
+import Alert from '../../common/components/Alert';
+import Switch from '../../common/components/Switch';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+import ItemList from '../../common/utils/ItemList';
+import icon from '../../common/helpers/icon';
+import saveSettings from '../utils/saveSettings';
+
+export default class AuthPage extends Page {
+  init() {
+    super.init();
+
+    this.saving = false;
+    this.refresh();
+  }
+
+  refresh() {
+    const settings = app.data.settings;
+
+    this.drivers = app.data.ssoDrivers;
+    this.driverFields = this.driverFieldsList().toArray();
+    this.driverInputs = [];
+
+    this.fields = ['allow_sign_up', 'enable_password_auth'];
+    this.values = {
+      allow_sign_up: m.prop(settings.allow_sign_up == '1'),
+      enable_password_auth: m.prop(settings.enable_password_auth == '1'),
+    };
+
+    for (const driver in this.drivers) {
+      this.driverFields.forEach(field => {
+        const fieldName = this.driverFieldKey(field.name, driver);
+
+        this.fields.push(fieldName);
+        this.values[fieldName] = m.prop(settings[fieldName]);
+
+        this.driverInputs[fieldName] = Checkbox.component({
+          state: this.values[fieldName](),
+          onchange: () => this.toggle(fieldName)
+        });
+      });
+    }
+
+    if (this.allSsoDisabled() && !this.values['enable_password_auth']()) {
+      this.values['enable_password_auth'](true);
+      this.saveSettings();
+      m.redraw();
```

Comment:
This `m.redraw()` is unnecessary as there's another just below outside of the `if`

Reply:
Fixed